multi-node/aws/pkg/config/config.go

```diff
@@ -403,29 +403,39 @@ func (cfg Cluster) valid() error {
 		return fmt.Errorf("invalid controllerIP: %s", cfg.ControllerIP)
 	}
 
+	var instanceCIDRs = make([]*net.IPNet, 0)
 	for i, subnet := range cfg.Subnets {
 		if subnet.AvailabilityZone == "" {
 			return fmt.Errorf("availabilityZone must be set for subnet #%d", i)
 		}
-		_, instancesNet, err := net.ParseCIDR(subnet.InstanceCIDR)
+		_, instanceCIDR, err := net.ParseCIDR(subnet.InstanceCIDR)
 		if err != nil {
 			return fmt.Errorf("invalid instanceCIDR for subnet #%d: %v", i, err)
 		}
-		if !vpcNet.Contains(instancesNet.IP) {
+		instanceCIDRs = append(instanceCIDRs, instanceCIDR)
+		if !vpcNet.Contains(instanceCIDR.IP) {
 			return fmt.Errorf("vpcCIDR (%s) does not contain instanceCIDR (%s) for subnet #%d",
 				cfg.VPCCIDR,
 				cfg.InstanceCIDR,
 				i,
 			)
 		}
-		if i == 0 && !instancesNet.Contains(controllerIPAddr) {
+		if i == 0 && !instanceCIDR.Contains(controllerIPAddr) {
 			return fmt.Errorf("instanceCIDR (%s) does not contain controllerIP (%s)",
 				subnet.InstanceCIDR,
 				cfg.ControllerIP,
 			)
 		}
 	}
 
+	for i, a := range instanceCIDRs {
+		for j, b := range instanceCIDRs[i+1:] {
+			if i > 0 && cidrOverlap(a, b) {
```

Comment:
Why do we need to check that `i > 0` here? If I'm reading this code correctly, it will always fail to check that `instanceCIDRs[0]` overlaps with `instanceCIDRs[1]`.